consensus/spos/bls/v2/subroundEndRound_test.go

```diff
@@ -684,99 +684,90 @@ func TestSubroundEndRound_ReceivedProof(t *testing.T) {
-			GetProofCalled: func(shardID uint32, headerHash []byte) (data.HeaderProofHandler, error) {
-				assert.Equal(t, hdr.GetPrevHash(), headerHash)
-				return &block.HeaderProof{
-					HeaderHash:          headerHash,
-					AggregatedSignature: providedPrevSig,
-					PubKeysBitmap:       providedPrevBitmap,
-				}, nil
-			},
-		})
-
-		ch := make(chan bool, 1)
-		consensusState := initializers.InitConsensusState()
-		sr, _ := spos.NewSubround(
-			bls.SrSignature,
-			bls.SrEndRound,
-			-1,
-			int64(85*roundTimeDuration/100),
-			int64(95*roundTimeDuration/100),
-			"(END_ROUND)",
-			consensusState,
-			ch,
-			executeStoredMessages,
-			container,
-			chainID,
-			currentPid,
-			&statusHandler.AppStatusHandlerStub{},
-		)
-
-		srEndRound, _ := v2.NewSubroundEndRound(
-			sr,
-			v2.ProcessingThresholdPercent,
-			&statusHandler.AppStatusHandlerStub{},
-			&testscommon.SentSignatureTrackerStub{},
-			&consensusMocks.SposWorkerMock{},
-			&dataRetrieverMocks.ThrottlerStub{},
-		)
+		container.SetBlockProcessor(bp)
+		sr := initSubroundEndRoundWithContainer(container, &statusHandler.AppStatusHandlerStub{})
+		_ = sr.SetJobDone(sr.SelfPubKey(), sr.Current(), true)
 
-		srEndRound.SetHeader(hdr)
-		srEndRound.AddReceivedHeader(hdr)
+		sr.ReceivedProof(&block.HeaderProof{})
+	})
+	t.Run("should early return when header is nil", func(t *testing.T) {
+		t.Parallel()
 
-		srEndRound.SetStatus(2, spos.SsFinished)
-		srEndRound.SetStatus(3, spos.SsNotFinished)
+		container := consensusMocks.InitConsensusCore()
+		bp := &testscommon.BlockProcessorStub{
+			CommitBlockCalled: func(header data.HeaderHandler, body data.BodyHandler) error {
+				require.Fail(t, "should have not been called")
+				return nil
+			},
+		}
+		container.SetBlockProcessor(bp)
+		sr := initSubroundEndRoundWithContainer(container, &statusHandler.AppStatusHandlerStub{})
+		sr.SetHeader(nil)
 
 		proof := &block.HeaderProof{}
-		srEndRound.ReceivedProof(proof)
+
+		sr.ReceivedProof(proof)
 	})
-	t.Run("should return false when header is nil", func(t *testing.T) {
+	t.Run("should early return when header is not for current consensus", func(t *testing.T) {
 		t.Parallel()
 
-		sr := initSubroundEndRound(&statusHandler.AppStatusHandlerStub{})
-		sr.SetHeader(nil)
+		hdr := &block.Header{Nonce: 37}
+		container := consensusMocks.InitConsensusCore()
+		bp := &testscommon.BlockProcessorStub{
+			CommitBlockCalled: func(header data.HeaderHandler, body data.BodyHandler) error {
+				require.Fail(t, "should have not been called")
+				return nil
+			},
+		}
+		container.SetBlockProcessor(bp)
+		sr := initSubroundEndRoundWithContainer(container, &statusHandler.AppStatusHandlerStub{})
+		sr.SetHeader(hdr)
+		sr.AddReceivedHeader(hdr)
 
 		proof := &block.HeaderProof{}
-
 		sr.ReceivedProof(proof)
 	})
-	t.Run("should return false when final info is not valid", func(t *testing.T) {
+	t.Run("should early return when final info is not valid", func(t *testing.T) {
```

Comment:
maybe rename to proof instead final info?

also it is not clear here which one fails, the leaderSignature is on the block, but the Signature is on the proof.\
In this test both return errors.